projects/step-frontend/src/lib/modules/timeseries/components/chart-dashlet/chart-dashlet.component.ts

```diff
@@ -159,7 +164,7 @@ export class ChartDashletComponent implements OnInit, Dashlet {
         data: series.map((b) => this.getBucketValue(b, primaryAggregation!)),
         value: (self, x) => TimeSeriesConfig.AXES_FORMATTING_FUNCTIONS.bigNumber(x),
         stroke: color,
-        fill: (self: uPlot, seriesIdx: number) => UPlotUtils.gradientFill(self, color),
+        fill: (self, seriesIdx: number) => this._uPlotUtils.gradientFill(self, color),
```

Comment:
I would prefer to avoid classes with static method at all.

If we have some functionality, that can be logically group in class, let the framework to manage how it should be created and where it should be accessible.

The things which can be placed separately (out of services) - some widely used functions, that doesn't connect to any specific entities, custom pipe operators, constants.

Reply:
I'm fine with both, but I think it's important to debate more on this idea (in general). Having your approach, makes you dependent on the framework (apart of the other performance implications). I see nowadays trend is to make independent (of the framework) code as much as possible, static classes being a very nice to have. @neogucky what's your opinion on this?

Reply:
I prefer to integrate with the framework in most cases. But regarding static functions I have no problem with them as they clearly show that data will only be transformed and not stored / etc. which can make the code cleaner in some ue cases. For this case I actually don't have a preference.